### DISCUSSION TITLE
sphinx-design: Sections in tabs ?

### DISCUSSION THREAD

**Discussion by varac** [executablebooks/meta#461] [Q&A]
Hi! I realize that non of my sections are rendered with sphinx-design in a tab. I wonder if this is by design, or if I'm doing anything wrong.

Here's an example:

```
=======
Testing
=======

.. tab-set::

    .. tab-item:: Label1

        Test
        ====
        Content 1

    .. tab-item:: Label2

        Content 2
        ---------
```

And I get this sphinx-build error:

```
Test
====
test.rst:16: WARNING: Unexpected section title
```

**Comment by chrisjsewell** [comment]
Heya, yeh this is mentioned in the TODO but I haven't opened a specific issue yet: https://github.com/executablebooks/sphinx-design#todo

sphinx requires a document structure to be hierarchical by heading levels:

```xml
<section>
  heading level 1

  <section>
     heading level 2

     <section>
         heading level 3
```

so you can't generally have headings inside directives, since this would mess up the structure:


```xml
<section>
  heading level 1

  <tab-container>
      <section>
         heading level 2

  <section>
      heading level 3
```

The main example of this being allowed is the `only` directive, where sphinx applies some "ad hoc" logic about how its parsed: https://github.com/sphinx-doc/sphinx/blob/8fd4373d3aec07b8d9a4fb159b6236f4dc715b21/sphinx/directives/other.py#L310

**Reply by varac** [reply]
Ok, good to know.
